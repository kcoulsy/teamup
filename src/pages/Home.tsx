import React from 'react';
import { Card, List, Typography } from 'antd';
import TeamInvites from '../components/TeamInvites/TeamInvites';

const Home = () => {
  return (
    <div>
      <TeamInvites />
      <Card title='Welcome!'>
        <p>
          Thanks for checking out the project, the intent of this project is to
          have a full stack application as an example for my development skills.
        </p>
        <p>
          The purpose of the app is to track Projects and Tasks, for personal
          use or to use as a team. With teams there is a whole role and
          permissions system that can be customised to the teams requirements.
        </p>
        <p>
          The project is far from finished, currently in <b>v0.1</b>. With
          plenty of features planned, I will update this demo as major additions
          are added. You can check out the project on github{' '}
          <a href='https://github.com/kcoulsy/teamup'>here</a>.
        </p>
        <List
          header={
            <Typography.Title level={4} style={{ margin: 0 }}>
              Planned Features
            </Typography.Title>
          }
          dataSource={[
            'Comments on tasks for teams',
            'Work log to track time spent on tasks',
            'Activity log across teams',
            'User and team profiles',
            'Mobile Resposiveness',
          ]}
          size='small'
          renderItem={(item) => <List.Item>{item}</List.Item>}></List>
      </Card>
    </div>
  );
};

export default Home;
