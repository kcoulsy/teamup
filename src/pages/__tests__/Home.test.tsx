import React from 'react';
import { shallow } from 'enzyme';
import { describe, expect, test, vi } from 'vitest';

import Home from './../Home';
import { shallowToJson } from 'enzyme-to-json';

describe('render Home component', () => {
  test('placeholder', () => {
    expect(true).toBe(true);
  });
  // TODO - update to react testing library
  //   test('should render Home page component', () => {
  //     const wrapper = shallow(<Home />);
  //     expect(shallowToJson(wrapper)).toMatchSnapshot();
  //   });
  //   // TODO - update to react testing library
  //   test('should allow modal to be toggled', () => {
  //     const wrapper = shallow(<Home />);
  //     expect(wrapper.find('Modal').prop('isActive')).toBe(true);
  //     wrapper.find('button').simulate('click', { preventDefault: vi.fn() });
  //     expect(wrapper.find('Modal').prop('isActive')).toBe(false);
  //   });
});
