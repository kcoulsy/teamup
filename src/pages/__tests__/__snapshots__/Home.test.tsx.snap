// Vitest Snapshot v1

exports[`render Home component should render Home page component 1`] = `
<div>
  <h3>
    Home Page
  </h3>
  <button
    className="ui button"
    onClick={[Function]}
  >
    Open Modal
  </button>
  <Modal
    actions={
      Array [
        Object {
          "buttonLabel": "Close",
          "klasses": "positive right labeled icon ",
          "onClick": [Function],
        },
      ]
    }
    headerTitle="hello"
    isActive={true}
  >
    this is content
  </Modal>
</div>
`;
